rust/rope/src/spans.rs

```diff
@@ -189,6 +195,15 @@ impl<T: Clone + Default> Spans<T> {
             ix: 0,
         }
     }
+
+    /// iterates through all spans, but in backwards order
```

Comment:
One interesting question about this: if you have nested spans, like `<a>...<b>...</b>...</a>`, are you returning `a` first or `b`, i.e. are you sorting by span begin or span end? Pretty sure it's the former, but probably worth documenting.